modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/IborCapletFloorletPeriodCurrencyAmounts.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (C) 2021 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.capfloor;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.product.capfloor.IborCapletFloorletPeriod;
+
+/**
+ * A map of currency amounts keyed by Ibor caplet/floorlet periods.
+ * <p>
+ * The currency amounts should represent the same metric.
+ */
+@BeanDefinition
+public final class IborCapletFloorletPeriodCurrencyAmounts implements ImmutableBean, Serializable {
```

Comment:
Is this only used for present values now? Might be more representative as IborCapletFloorletPeriodPresentValue?

Reply:
Right now it is only used for present values, but I think it's more consistent this way with `IborCapletFloorletPeriodAmounts` which has doubles, but can be forward rates/implied vols. By keeping this name, it allows any currency amount to be stored against the periods.